hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/store/contracts/precompile/TransferPrecompilesStubTest.java

```diff
@@ -0,0 +1,586 @@
+                new ChangeSummaryManager<>());
+        final var liveTokens = new TransactionalLedger<>(
+                TokenProperty.class, MerkleToken::new, new HashMapBackingTokens(), new ChangeSummaryManager<>());
+        final FCHashMap<ByteString, EntityNum> aliases = new FCHashMap<>();
+        final var liveAliases = new AliasManager(() -> aliases);
+        final var ledgers = new WorldLedgers(liveAliases, liveTokenRels, liveAccounts, liveNfts, liveTokens);
+
+        // The changes passed in here are not necessarily the same as the changes derived from the input bytes.
+        // Stubbing impliedTransfersMarshal.assessCustomFeesAndValidate without mocking was too involved.
+        // Instead, various alternative changes are passed in as parameters.
+        final ImpliedTransfers impliedTransfers =
+                ImpliedTransfers.valid(validationProps, changes, Collections.emptyList(), Collections.emptyList());
+
+        given(infrastructureFactory.newImpliedTransfersMarshal(any())).willReturn(impliedTransfersMarshal);
+        given(infrastructureFactory.newHederaTokenStore(sideEffects, liveTokens, liveNfts, liveTokenRels))
+                .willReturn(hederaTokenStore);
+        given(infrastructureFactory.newTransferLogic(
+                        hederaTokenStore, sideEffects, liveNfts, liveAccounts, liveTokenRels))
+                .willReturn(transferLogic);
+        given(impliedTransfersMarshal.validityWithCurrentProps(any())).willReturn(OK);
+        given(impliedTransfersMarshal.assessCustomFeesAndValidate(anyInt(), anyInt(), anyInt(), any(), any(), any()))
+                .willReturn(impliedTransfers);
+        // This test does not cover key activation. Rather a parameter is passed in to indicate pass/fail.
+        keyActivationUtils
+                .when(() -> validateKey(any(), any(), any(), any(), any(), eq(CryptoTransfer)))
+                .thenReturn(passesKeyActivation);
+
+        subject = new TransferPrecompile(
+                ledgers,
+                worldUpdater,
+                sigsVerifier,
+                sideEffects,
+                syntheticTxnFactory,
+                infrastructureFactory,
+                precompilePricingUtils,
+                selector,
+                senderAddress,
+                htsUnsupportedCustomFeeReceiverDebits, // This is passed in as a parameter
+                true,
+                false,
+                canFallbaToApprovals);
+
+        if (bodyMethodShouldPass) {
+            subject.body(inputBytes, a -> a);
+        } else {
+            assertThrows(IllegalArgumentException.class, () -> subject.body(inputBytes, a -> a));
+            return;
+        }
+
+        if (runMethodShouldPass) {
+            subject.run(frame);
+        } else {
+            assertThrows(InvalidTransactionException.class, () -> subject.run(frame));
+        }
+    }
+
+    private static Stream<Arguments> provideTransferPrecompileArgs() {
```

Comment:
_Excellent_ use of parameterized test here.  

Only quibble I'd make is that there could be enum classes (with two values `YES` and `NO`) for each of the 4 booleans - given named enum values these test cases could be read right off this setup here without having to keep track of which `true` or `false` was which.